File: Medium/JumpGame.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let furthest = 0;
    let lastIndex = nums.length - 1;
    for (let i = 0; i <= furthest && furthest <= lastIndex; i++) {
        furthest = Math.max(furthest, i + nums[i]);
        if (furthest >= lastIndex) {
            return true;
        }
    }
    return furthest >= lastIndex;
};

File: Medium/JumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        furthest = 0
        last_index = len(nums) - 1

        for index, item in enumerate(nums):
            if index > furthest:
                return False
            furthest = max(furthest, index + item)
            

        return furthest >= last_index
